topology/src/main/java/io/camunda/zeebe/topology/changes/PartitionReconfigurePriorityApplier.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.changes;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.topology.changes.TopologyChangeAppliers.OperationApplier;
+import io.camunda.zeebe.topology.state.ClusterTopology;
+import io.camunda.zeebe.topology.state.MemberState;
+import io.camunda.zeebe.topology.state.MemberState.State;
+import io.camunda.zeebe.topology.state.PartitionState;
+import io.camunda.zeebe.util.Either;
+import java.util.function.UnaryOperator;
+
+public class PartitionReconfigurePriorityApplier implements OperationApplier {
+
+  private final int partitionId;
+  private final int newPriority;
+  private final MemberId localMemberId;
+  private final PartitionChangeExecutor partitionChangeExecutor;
+
+  public PartitionReconfigurePriorityApplier(
+      final int partitionId,
+      final int priority,
+      final MemberId memberId,
+      final PartitionChangeExecutor partitionChangeExecutor) {
+    this.partitionId = partitionId;
+    newPriority = priority;
+    localMemberId = memberId;
+    this.partitionChangeExecutor = partitionChangeExecutor;
+  }
+
+  @Override
+  public Either<Exception, UnaryOperator<MemberState>> init(
```

Comment:
:thought_balloon: Probably not necessary but we _could_ validate that there are no duplicate priorities.

Reply:
There will be duplicate priorities while we are applying the changes because we are changing it one by one.